caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/NotificationAttachmentService.java

```diff
@@ -0,0 +1,137 @@
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.laa.ccms.caab.api.entity.NotificationAttachment;
+import uk.gov.laa.ccms.caab.api.exception.CaabApiException;
+import uk.gov.laa.ccms.caab.api.mapper.NotificationAttachmentMapper;
+import uk.gov.laa.ccms.caab.api.repository.NotificationAttachmentRepository;
+import uk.gov.laa.ccms.caab.model.NotificationAttachmentDetail;
+import uk.gov.laa.ccms.caab.model.NotificationAttachmentDetails;
+
+/**
+ * Service responsible for handling notification attachment operations.
+ *
+ * @see NotificationAttachmentRepository
+ * @see NotificationAttachmentMapper
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class NotificationAttachmentService {
+
+  private final NotificationAttachmentRepository repository;
+  private final NotificationAttachmentMapper mapper;
+
+  /**
+   * Get a Page of NotificationAttachments for the supplied search criteria.
+   *
+   * @param providerId - the provider id.
+   * @param documentType - the document type.
+   * @param pageable - the pageable settings.
+   * @return NotificationAttachmentDetails wrapping a page of NotificationAttachments.
+   */
+  public NotificationAttachmentDetails getNotificationAttachments(
+      final String notificationReference,
+      final String providerId,
+      final String documentType,
+      final String sendBy,
+      final Pageable pageable) {
+
+    Example<NotificationAttachment> exampleDocument =
+        buildExampleDocument(notificationReference, providerId, documentType, sendBy);
+
+    return mapper.toNotificationAttachmentDetails(repository.findAll(exampleDocument, pageable));
+  }
+
+  /**
+   * Get a single NotificationAttachmentDetail by id.
+   *
+   * @param id - the notification attachment id.
+   * @return NotificationAttachmentDetail with the matching id, or else an error.
+   */
+  public NotificationAttachmentDetail getNotificationAttachment(final Long id) {
+
+    return repository
+        .findById(id)
+        .map(mapper::toNotificationAttachmentDetail)
+        .orElseThrow(
+            () ->
+                new CaabApiException(
+                    String.format("Failed to find notification attachment with id: %s", id),
+                    HttpStatus.NOT_FOUND));
+  }
+
+  /**
+   * Create a new NotificationAttachment entry.
+   *
+   * @param notificationAttachmentDetail - the notification attachment to create.
+   * @return the id of the newly created notification attachment.
+   */
+  public Long createNotificationAttachment(
+      final NotificationAttachmentDetail notificationAttachmentDetail) {
+
+    uk.gov.laa.ccms.caab.api.entity.NotificationAttachment createdNotificationAttachment =
```

Comment:
Class is imported, so no need for fully qualified?

Reply:
Fixed